sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JSONOptions.scala

```diff
@@ -68,6 +68,8 @@ private[sql] class JSONOptions(
     parameters.get("allowNonNumericNumbers").map(_.toBoolean).getOrElse(true)
   val allowBackslashEscapingAnyCharacter =
     parameters.get("allowBackslashEscapingAnyCharacter").map(_.toBoolean).getOrElse(false)
+  val allowDateFormat =
+    parameters.get("allowDateFormat").map(_.toBoolean).getOrElse(false)
```

Comment:
Shall we add this option into CSV too? I would name it like `inferDateType`. BTW, I am just curious. Why did you disable it specifically here in JSON?

Reply:
We would also have to add the docs in https://github.com/apache/spark/blob/master/docs/sql-data-sources-json.md and https://github.com/apache/spark/blob/master/docs/sql-data-sources-csv.md

Reply:
**Shall we add this option into CSV too? I would name it like inferDateType.** - Added for CSV also and change the name to inferDateType

 **BTW, I am just curious. Why did you disable it specifically here in JSON?** - This is to keep in sync with older version of spark, If someone wants to use the dateType, they can enable it in the option. This is just to prevent any migration issue to spark-3.2.0 from the older version.

Reply:
**We would also have to add the docs in https://github.com/apache/spark/blob/master/docs/sql-data-sources-json.md and https://github.com/apache/spark/blob/master/docs/sql-data-sources-csv.md** - Added the doc for the option in the sql-data-sources-json.md but not able to find place to add the option in sql-data-sources-csv.md